cmd/argocd/commands/admin/project_allowlist.go

```diff
@@ -78,23 +81,38 @@ func NewProjectAllowListGenCommand() *cobra.Command {
 	return command
 }
 
-func generateProjectAllowList(clientConfig clientcmd.ClientConfig, clusterRoleFileName string, projName string) v1alpha1.AppProject {
+func getResourceList(clientConfig clientcmd.ClientConfig) ([]*metav1.APIResourceList, error) {
+	config, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, err
```

Comment:
As we change code that just return `err` I think we should start adding more context to better understand exceptional scenarios in logs. Example:
```go
return nil, fmt.Errorf("error while creating client config: %s", err)
```